app-frontend/src/app/components/histogram/dataVizHistogram/index.js

```diff
@@ -0,0 +1,114 @@
+    }
+
+    $postLink() {
+        this.isLoadingHistogram = true;
+        this.$q.all({
+            histStat: this.histogram,
+            histBounds: this.bounds
+        }).then(({histStat, histBounds}) => {
+            const {histogram, statistics} = histStat;
+            if (histogram && histBounds && get(histogram, 'buckets.length')) {
+                this.$scope.$evalAsync(() => this.initGraph(histogram, histBounds));
+                this.getGraph().then(graph => {
+                    this.createPlotFromHistogram(histogram, histBounds, statistics);
+                    graph.setData({histogram: this.plot})
+                        .render();
+                    this.isLoadingHistogram = false;
+                });
+            }
+        });
+    }
+
+    initGraph(histogram, bounds) {
+        this.options = {
+            dataRange: {
+                min: bounds.min,
+                max: bounds.max
+            },
+            height: 50,
+            style: {
+                height: '50px'
+            },
+            type: 'stat',
+            margin: {
+                top: 0,
+                bottom: 0,
+                right: 0,
+                left: 0
+            }
+        };
+
+        const elem = this.$element.find('.graph-container svg')[0];
+        this.graph = this.graphService.register(elem, this.id, this.options);
+    }
+
+    getGraph() {
+        return this.graphService.getGraph(this.id);
+    }
+
+    createPlotFromHistogram(histogram, bounds, statistics) {
+        const diff = (bounds.max - bounds.min) / 100;
+        const magnitude = Math.round(Math.log10(diff));
+        this.precision = Math.pow(10, magnitude);
+
+        const buckets = histogram.buckets;
+        let plot;
+        if (buckets.length < 100) {
+            let valueMap = new Map();
+            buckets.forEach(([bucket, value]) => {
+                const roundedBucket = Math.round(bucket / this.precision) * this.precision;
+                valueMap = valueMap.set(roundedBucket, value);
+            });
+            range(0, 100).forEach((mult) => {
+                const key = Math.round(
+                    (mult * diff + this.options.dataRange.min) / this.precision
+                ) * this.precision;
+                const value = valueMap.get(key, 0);
+                valueMap = valueMap.set(key, value);
+            });
+
+            plot = valueMap.entrySeq()
+                .sort(([K1], [K2]) => K1 - K2)
+                .map(([key, value]) => ({
+                    x: min([max([key, histogram.minimum]), histogram.maximum]),
+                    y: value}))
+                .toArray();
+        } else {
+            plot = buckets.map((bucket) => {
+                return {
+                    x: Math.round(bucket[0]), y: bucket[1]
+                };
+            });
+        }
+
+        this.plot = plot;
+    }
+}
+
+const component = {
+    bindings: {
+        id: '<',
```

Comment:
We should randomly generate graph ids in the dataVizHistogram component instead of taking it as a parameter. That could easily end up causing confusing errors for us later on, and the only thing we use the id for is uniquely identifying the graph.

Reply:
@Lknechtli agreed. I am passing in `trackId` as mentioned in the above conversation to the component instead.